tests/unit/tracing/test_unit_agent_trace_duration.py

```diff
@@ -36,6 +36,11 @@ def test_agent_trace_duration_from_sample(agent_trace: AgentTrace) -> None:
     changes, this test will need to be updated
     (because it is using start and end times that were manually parsed from the trace)
     """
-    expected_seconds = (1747660970774416000 - 1747660964057285000) / 1_000_000_000
+    # grab the start and end times from the last span in the trace
+    start_time = agent_trace.spans[-1].start_time
+    end_time = agent_trace.spans[-1].end_time
```

Comment:
This way we don't need to update the test every time the trace file changes, and we also have some sort of different logic from the .duration property (which is checking for the any_agent.run_id attribute). If that attribute span is no longer last, this test will now also tell us that broken assumption.

Reply:
I'm not sure I follow, but are you making assumptions on the trace ordering?

Reply:
> If that attribute span is no longer last, this test will now also tell us that broken assumption.

This is a good assumption to have in tests, specially once I finish #249